qt/python/mantidqt/widgets/workspacedisplay/table/presenter.py

```diff
@@ -177,16 +180,20 @@ def handleItemChanged(self, item: QStandardItem):
         :type item: A reference to the item that has been edited
         """
         try:
-            self.model.set_cell_data(item.row(), item.column(), item.data(Qt.DisplayRole),
```

Comment:
My thinking when writing this was that `handleItemChanged` would not be called for items that weren't editable meaning that any item here was guaranteed to be a `RevertibleItem`. It would seem that this method is still called for deletions? Is it possible to just stop that?

Reply:
If we can't stop the `handleItemChanged` call can we base the check on either a `hasattr` check on `item` of an `isintance(item, RevertibleItem)` rather than specifically being a peaks workspace? If something else wants to implement the table display interface then it would be good not to have further type checks in there.

Reply:
@martyngigg 
My understanding is that _deletion_ is a modification on items within the selection, and my local test shows that the deletion is performed on a per-item base, e.g. items within the selection (row) is deleted one at a time. Therefore, `handleItemChanged` is triggered upon deletion makes sense to me.

Also, this signal is connected through this line 
https://github.com/mantidproject/mantid/blob/5abc57f6ac867763705da3dab0d4193127894339/qt/python/mantidqt/widgets/workspacedisplay/table/presenter.py#L150
which means the connection is defined at the model level (the whole table) instead of individual cell. So I am afraid there is no easy way to provide different type of connection for each item within the same Table.

That being said, we could do an `isinstance` check at the beginning of `handleItemChanged`
```python
def handleItemChanged(self, item: QStandardItem):
    if not isinstance(item, RevertibleItem):
        pass
    ...
```
which effectively makes `handleItemChanged` a dummy function for items that is not derived from `RevertibleItem`. However, it is worth pointing out that the overhead from the function call and the instance check are unavoidable, so some level of performance penalty will still be there, especially for selections with large number of items.

Reply:
@KedoKudo Thanks for the explanation - that makes sense. I think the overhead of checking this is most likely minimal compared to painting the GUI or removing the items from vectors under the hood so I doubt we need to worry about it. As long as the GUI is still responsive then that's fine. Just one question - do you mean `return` rather than `pass` above:
```
def handleItemChanged(self, item: QStandardItem):
    if not isinstance(item, RevertibleItem):
        return
```

Reply:
yep, it should have been `return` instead of `pass`.